backend/beyond_the_loop/routers/auths.py

```
@@ -548,8 +564,21 @@ async def signout(request: Request, response: Response):
+        password = ''.join(secrets.choice(alphabet) for _ in range(length))
+        # Check if password has at least one of each: uppercase, lowercase, digit, special char
+        if (any(c.isupper() for c in password)
+            and any(c.islower() for c in password)
+            and any(c.isdigit() for c in password)
+            and any(c in "!@#$%^&*" for c in password)):
+            return password
+
+
 @router.post("/add", response_model=SigninResponse)
-async def add_user(form_data: AddUserForm, user=Depends(get_admin_user)):
+async def add_user(form_data: AddUserForm, admin_user: Users = Depends(get_admin_user)):
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)